src/common/version/build.rs

```diff
@@ -19,7 +19,7 @@ use build_data::{format_timestamp, get_source_time};
 use shadow_rs::{CARGO_METADATA, CARGO_TREE};
 
 fn main() -> shadow_rs::SdResult<()> {
-    println!("cargo:rerun-if-changed=.git/refs/heads");
+    println!("cargo:rerun-if-changed=../../../.git/refs/heads");
```

Comment:
Why this is a fix?

Reply:
Found https://github.com/rust-lang/cargo/issues/4213

Reply:
I'm not very familiar with the gix library and not sure if it will work with git submodules.

It's best to keep it simple. I can improve it once I'm sure it works.

Reply:
Record a code snippet for handling submodule:

```rust
fn main() {
    let mut current = std::path::PathBuf::new().join(env!("CARGO_MANIFEST_DIR"));

    while let Ok((dir, _)) = gix_discover::upwards(&current) {
        let dir = match dir {
            gix_discover::repository::Path::LinkedWorkTree { work_dir, .. } => {
                current = work_dir.parent().unwrap().to_path_buf();
                continue;
            }
            gix_discover::repository::Path::WorkTree(dir) => dir,
            gix_discover::repository::Path::Repository(dir) => dir,
        };

        let git_refs_heads = dir.join(".git/refs/heads");
        println!("cargo::rerun-if-changed={}", git_refs_heads.display());
        break;
    }
}
```

@Byron hopefully this is the correct way to "discover the git repository path that is not a submodule".

Reply:
I just re-read the [API docs](https://github.com/Byron/gitoxide/blob/e9295dc7a98eaedddb6941daa39b80412164360d/gix-discover/src/repository.rs#L6-L13) and could only see that it *can* be a submodule, but it could also be another worktree checkout.

`gix-discover` can only give an idea of what it probably is, and to know for sure one would have to open the repository and perform more checks.

However, with modern Git repositories, the `git_dir` of `LinkedWorkTree` should look like `…/modules/<name>`, so one could additionally try to find the `modules` subdirectory to be reasonably sure it's a submodule worktree you are looking at.